devito/interfaces.py

```diff
@@ -255,3 +265,60 @@ def indices(cls, shape):
         """
         _indices = [t, x, y, z]
         return _indices[:len(shape) + 1]
+
+
+class MemmapManager():
+    """Class for managing all memmap related settins"""
+    # used to enable memmap as default
+    _use_memmap = False
+    # flag for registering exit func
+    _registered = False
+    _default_disk_path = "/tmp/devito_disk"
+    # contains str name of all memmap file created
+    _created_data = set()
+    _default_exit_code = 0
+
+    @staticmethod
+    def memmap_enabled_by_default():
+        """call this method to enable memmap by default"""
+        return MemmapManager._use_memmap
+
+    @staticmethod
+    def set_default_disk_path(default_disk_path):
+        """call this method to change the default disk path for memmap"""
+        MemmapManager._default_disk_path = default_disk_path
+
+    @staticmethod
+    def setup(data_self, *args, **kwargs):
+        """call this method with arguments self, args, kwargs to setup memmap parameters"""
+        disk_path = kwargs.get('disk_path', MemmapManager._default_disk_path)
+        if not os.path.exists(disk_path):
+            os.makedirs(disk_path)
+        data_self.f = disk_path + "/data_" + kwargs.get('name')
+        data_self.memmap = kwargs.get('memmap', False) or MemmapManager._use_memmap
+        MemmapManager._created_data.add(data_self.f)
+        if not MemmapManager._registered:
+            MemmapManager._register_remove_memmap_file_signal()
+            MemmapManager._registered = True
+
+    # clean up memmap file
+    @staticmethod
+    def _reomve_memmap_file():
+        for f in MemmapManager._created_data:
+            try:
+                os.remove(f)
+            except OSError:
+                print("error removing " + f + " skipping")
+                pass
+
+    # used for clean memmap file on signal, internal method
+    @staticmethod
+    def _remove_memmap_file_on_signal(*args):
```

Comment:
Hi, so because we want to do 2 things: 1. clean up file on normal termination. 2. clean up on kill or interrupt. atexit is for 1. For 2, we trap the signals, but because we trapped the signal, we are also responsible for doing the exit. There are 2 was of exiting that I am aware of(could be more), one is to use sys.exit() which will trigger the func registered with atexit, so this is where we are doing the clean up. Since it already triggered cleanup, we do not want to call_reomve_memmap_file() again.  Aonther way is to call os._exit(), which does not trigger _reomve_memmap_file(). However, from what I read online, it is not common to call os._exit() directly. Thanks.